### PR TITLE
planner: fix idxMergePartPlans forget to deal with RootTaskConds

### LINKED ISSUE #58476: Index merge wrongly ignore some predicates of generated column
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
CREATE TABLE t3 (
  id int PRIMARY KEY,
  c1 varchar(256),
  c2 varchar(256) GENERATED ALWAYS AS (concat(c1, c1)) VIRTUAL,
  KEY (id)
);
 
insert into t3(id, c1) values (50, 'c');

SELECT /*+ USE_INDEX_MERGE(`t3`)*/ id FROM `t3` WHERE c2 BETWEEN 'a' AND 'b' GROUP BY id HAVING id < 100 or id > 0;
```

### 2. What did you expect to see? (Required)

```sql
Empty set
```

### 3. What did you see instead (Required)

```sql
mysql> SELECT /*+ USE_INDEX_MERGE(`t3`)*/ id FROM `t3` WHERE c2 BETWEEN 'a' AND 'b' GROUP BY id HAVING id < 100 or id > 0;
+----+
| id |
+----+
| 50 |
+----+
1 row in set (0.01 sec)
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #58476

Problem Summary:

### What changed and how does it work?

fix idxMergePartPlans forget to deal with RootTaskConds

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix idxMergePartPlans forget to deal with RootTaskConds

修复在处理 idxMergePartPlans 时忘记处理 RootTaskConds 的问题
```

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

File: pkg/planner/core/issuetest/planner_issue_test.go [hunk 1]
```diff
@@ -159,3 +159,21 @@ func TestIssues57583(t *testing.T) {
...
 }
+
+func TestIssue58476(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test;")
+	tk.MustExec("CREATE TABLE t3 (id int PRIMARY KEY,c1 varchar(256),c2 varchar(256) GENERATED ALWAYS AS (concat(c1, c1)) VIRTUAL,KEY (id));")
+	tk.MustExec("insert into t3(id, c1) values (50, 'c');")
+	tk.MustQuery("SELECT /*+ USE_INDEX_MERGE(`t3`)*/ id FROM `t3` WHERE c2 BETWEEN 'a' AND 'b' GROUP BY id HAVING id < 100 or id > 0;").Check(testkit.Rows())
+	tk.MustQuery("EXPLAIN SELECT /*+ USE_INDEX_MERGE(`t3`)*/ id FROM `t3` WHERE c2 BETWEEN 'a' AND 'b' GROUP BY id HAVING id < 100 or id > 0;").
```
Comment (qw4990) [inline pkg/planner/core/issuetest/planner_issue_test.go:170]: Better to use "explain format='brief'"

Comment (hawkingrei) [reply pkg/planner/core/issuetest/planner_issue_test.go]: OK, it has been updated.

**Review by winoros — APPROVED** [review]

**Comment by hawkingrei** [comment]
/hold

self-check

**Pull request merged**